Trappist/src/Promact.Trappist.Web/wwwroot/app/profile/profile-dashboard/change-password-dialog.component.ts

```diff
@@ -1,8 +1,48 @@
-﻿import { Component } from "@angular/core";
+﻿import { Component } from '@angular/core';
+import { ChangePasswordModel } from '../password.model';
+import { MdDialogRef } from '@angular/material';
+import { ProfileService } from '../profile.service';
+import { MdSnackBar } from '@angular/material';
 
 @Component({
-  moduleId: module.id,
-  selector: 'change-password-dialog',
-  templateUrl: "change-password-dialog.html"
+    moduleId: module.id,
+    selector: 'change-password-dialog',
+    templateUrl: 'change-password-dialog.html'
 })
-export class ChangePasswordDialogComponent { }
+export class ChangePasswordDialogComponent {
+    constructor(public profileService: ProfileService, public dialog: MdDialogRef<any>, public snackBar: MdSnackBar) { }
+    user: ChangePasswordModel = new ChangePasswordModel();
+    isMessageMismatch: boolean = true;
+    messageMismatch: string;
+    response: any;
+    errorMesseage: any;
+    errorCorrection: boolean = true;
+
+    /**
+     * update the database with new password
+     * @param userPassword of type ChangePasswordModel which has the new password of the user
+     */
+    changePassword(userPassword: ChangePasswordModel) {
+        if (userPassword.oldPassword !== null && userPassword.oldPassword !== ' ' && userPassword.oldPassword !== undefined && userPassword.newPassword !== null && userPassword.newPassword !== ' ' && userPassword.newPassword !== undefined && userPassword.confirmPassword !== null && userPassword.confirmPassword !== ' ' && userPassword.confirmPassword !== undefined) {
+            if (userPassword.newPassword === userPassword.confirmPassword && userPassword.confirmPassword === userPassword.newPassword) {
```

Comment:
if (userPassword.newPassword === userPassword.confirmPassword && userPassword.confirmPassword === userPassword.newPassword)

Ok, what will happen if you don't write userPassword.confirmPassword === userPassword.newPassword line ?